File: backend/app/models/memory.py
import uuid
from typing import Optional

from sqlalchemy import Boolean, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, TimestampMixin


class Memory(Base, TimestampMixin):
    __tablename__ = "memories"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    content: Mapped[str] = mapped_column(Text)
    category: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, server_default="1")

File: frontend/src/api/memory.ts
import client from './client'

export interface Memory {
  id: string
  content: string
  category: string | null
  is_active: boolean
  created_at: string
}

export async function getMemories(): Promise<Memory[]> {
  const { data } = await client.get<Memory[]>('/memories')
  return data
}

export async function deleteMemory(id: string): Promise<void> {
  await client.delete(`/memories/${id}`)
}
